numpy/_core/strings.py

```diff
@@ -652,33 +657,31 @@ def ljust(a, width, fillchar=' '):
     array(['aAa', '  a', 'abB'], dtype='<U3')
     
     """
-    a_arr = np.asarray(a)
-    width_arr = np.asarray(width)
-    size = int(np.max(width_arr.flat))
-    if np.issubdtype(a_arr.dtype, np.bytes_):
-        fillchar = np._utils.asbytes(fillchar)
-    if isinstance(a_arr.dtype, np.dtypes.StringDType):
-        res_dtype = a_arr.dtype
+    a = np.asanyarray(a)
+    width = np.maximum(str_len(a), width)
+    fillchar = np.asanyarray(fillchar)
+
+    if a.dtype.char == "T":
+        shape = np.broadcast_shapes(a.shape, width.shape, fillchar.shape)
+        out = np.empty_like(a, shape=shape)
```

Comment:
Not really so relevant here, but I'm actually a bit surprised we cannot just do `out=None` for `StringDType`...